mcodingbot/tasks/stats.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+import asyncio
+import traceback
+from dataclasses import dataclass
+from math import log
+from typing import TYPE_CHECKING, Any
+
+from mcodingbot.config import CONFIG
+
+if TYPE_CHECKING:
+    from mcodingbot.bot import Bot
+
+
+async def loop_update_channels(bot: Bot) -> None:
+    while True:
+        try:
+            await update_channels(bot)
+        except Exception:
+            print("Failed to update channel stats:")
+            traceback.print_exc()
+
+        await asyncio.sleep(5 * 60)
+
+
+async def update_channels(bot: Bot) -> None:
+    stats = await get_stats(bot)
+
+    sub_channel = bot.cache.get_guild_channel(CONFIG.sub_count_channel)
+    view_channel = bot.cache.get_guild_channel(CONFIG.view_count_channel)
+    member_channel = bot.cache.get_guild_channel(CONFIG.member_count_channel)
+
+    assert sub_channel
+    assert view_channel
+    assert member_channel
+
+    await sub_channel.edit(name=f"Subs: {display_stats(stats.subs)}")
+    await view_channel.edit(name=f"Views: {display_stats(stats.views)}")
+
+    guild = bot.cache.get_guild(CONFIG.mcoding_server)
+    assert guild
+    member_count = guild.member_count
+    assert member_count is not None
+
+    await member_channel.edit(name=f"Members: {display_stats(member_count)}")
+
+
+@dataclass
+class Stats:
+    subs: float
+    views: float
+
+
+BASE_URL = "https://www.googleapis.com/youtube/v3/channels"
+LAST_STATS = Stats(0, 0)
```

Comment:
I feel like `MOST_RECENT_STATS` is a little clearer. Also i dont think this is a constant so it should be regular snake case.